dd-sdk-android-ndk/src/main/cpp/utils/signal-monitor.c

```diff
@@ -80,17 +76,33 @@ static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
 
 size_t handled_signals_size() { return sizeof(handled_signals) / sizeof(handled_signals[0]); }
 
-// It may happen that the process is killed during this function execution.
-// Therefore this function may never return.
-void invoke_previous_handler(int signum, siginfo_t *info, void *user_context) {
 
+void free_up_memory() {
+    if (datadog_sigaction != NULL) {
+        free(datadog_sigaction);
+    }
+    if (original_sigaction != NULL) {
+        free(original_sigaction);
+    }
+    if (signal_stack.ss_sp != NULL) {
+        free(signal_stack.ss_sp);
+    }
+    datadog_sigaction = NULL;
```

Comment:
Put each of these inside their respective `if` block, makes it easier to see that we haven't missed any if we add more to this later